include/boost/json/detail/impl/format.ipp

```diff
@@ -111,10 +111,44 @@ format_int64(
 
 unsigned
 format_double(
-    char* dest, double d) noexcept
+    char* dest, double d, const serialize_options & opt) noexcept
 {
-    return static_cast<int>(
-        ryu::d2s_buffered_n(d, dest));
+    if (std::isnan(d))
+    {
+        std::copy(opt.nan.begin(), opt.nan.end(), dest);
+        return opt.nan.size();
+    }
+    else if (std::isinf(d))
+    {
+        if (d > 0.)
+        {
+            std::copy(opt.pinf.begin(), opt.pinf.end(), dest);
```

Comment:
This code has to handle the case where there is not enough room in the output to hold the user-defined string